relayer/src/channel.rs

```diff
@@ -622,33 +639,46 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Channel<ChainA, ChainB> {
         .map_err(|e| ChannelError::query_channel(channel_id.clone(), e))
     }
 
-    pub fn handshake_step(&mut self, state: State) -> Result<Vec<IbcEvent>, ChannelError> {
-        match (state, self.counterparty_state()?) {
-            (State::Init, State::Uninitialized) => Ok(vec![self.build_chan_open_try_and_send()?]),
-            (State::Init, State::Init) => Ok(vec![self.build_chan_open_try_and_send()?]),
-            (State::TryOpen, State::Init) => Ok(vec![self.build_chan_open_ack_and_send()?]),
-            (State::TryOpen, State::TryOpen) => Ok(vec![self.build_chan_open_ack_and_send()?]),
-            (State::Open, State::TryOpen) => Ok(vec![self.build_chan_open_confirm_and_send()?]),
-            _ => Ok(vec![]),
-        }
-    }
+    pub fn handshake_step(
+        &mut self,
+        state: State,
+    ) -> Result<(Option<IbcEvent>, bool), ChannelError> {
```

Comment:
Can we replace this boolean with an enum which describes whether the handshake is completed or not?

eg. 

```rust
enum HandshakeStatus {
  Completed,
  InProgress
}
```

Reply:
I make it return `task::Next` directly instead.